src/reducers/AuthReducer.js

```diff
@@ -24,16 +47,46 @@ export default (state = INITIAL_STATE, action) => {
             return ({...state, appReady: true});
 
         case GUEST_TOKEN_SET:
-            return ({...state, token: payload.token, appReady: payload.appReady});
+            const guestUser = {name: payload.token};
+            return ({...state, token: payload.token, currentUser: guestUser, appReady: payload.appReady});
 
         case FACEBOOK_LOGIN_SUCCESS:
-            return ({...state, token: payload.token, appReady: payload.appReady});
+            return ({...state, token: payload.token, currentUser: payload.currentUser, appReady: payload.appReady});
 
         case FACEBOOK_LOGIN_FAILURE:
             return ({...state, token: null});
 
         case FACEBOOK_LOGOUT_COMPLETE:
-            return ({...state, token: null, appReady: false});
+            return (INITIAL_STATE);
+
+        case FAVORITE_ADDED:
+            const {favoriteToAdd} = payload;
+
+            let updatedUserWithNewFavorite = _.cloneDeep(state.currentUser);
+            updatedUserWithNewFavorite.favorites = _.concat(updatedUserWithNewFavorite.favorites, favoriteToAdd);
+
+            return ({...state, currentUser: updatedUserWithNewFavorite});
+
+        case FAVORITE_REMOVED:
+            const {favoriteToRemove} = payload;
+            let updatedUserWithoutFavorite = _.cloneDeep(state.currentUser);
+            updatedUserWithoutFavorite.favorites = _.reject(updatedUserWithoutFavorite.favorites, favorite => favorite.id === favoriteToRemove.id);
+
+            return ({...state, currentUser: updatedUserWithoutFavorite});
```

Comment:
When a favorite is removed, remove it from the local list. 

One thing to keep in mind about both of these add/remove actions is that they are only triggered when we get a `success` message from firebase that we also took care of the remote work on the database, so we don't have to worry about our local instance falling out of sync with the remote db.